src/service.rs

```diff
@@ -86,78 +89,64 @@ where
                             }
                         };
 
-                        (AxumSessionID(new_id), false)
-                    };
-
-                    // If a cookie did have an AxumSessionID then lets check if it still exists in the hash or Database
-                    // If not make a new Session using the ID.
-                    if id.1 {
-                        let mut no_store = true;
+                        is_new_id = true;
+                        AxumSessionID(new_id)
+                    }
+                },
+                store: store.clone(),
+            };
 
-                        {
-                            if let Some(m) = store.inner.read().await.get(&id.0.to_string()) {
-                                let mut inner = m.lock().await;
+            let accepted = if let Some(cookie) = cookies.get(&store.config.accepted_cookie_name) {
+                cookie.value().parse().unwrap_or(false)
+            } else {
+                false
+            };
```

Comment:
This reads a bit weird. If you're starting with an `if let Some(..)` can't you just chain map/and_then all the way? You're already unwraping to false which matches the else-case.